MTM-62313][MTM-62316] Dashboard details Import/Export and new approach as breaking change announcement. (#3223)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1022-0-0-dashboard-cross-application-copy-paste.md

---
date: 2025-04-16
title: Reports page extracted into a separate plugin
product_area: Application enablement & solutions
change_type:
  - value: change-inv-3bw8e
    label: Announcement
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-62313, MTM-62316
version: 1022.0.0
---

> In the upcoming version dashboard settings component will be refactored to use secondary router outlet.

Reviewer: ```suggestion
In an upcoming version the dashboard settings component will be refactored to use a secondary router outlet.
```

> This approach allows to hook new tab to particular outlet, e.g. 

Reviewer: ```suggestion
This approach allows to hook a new tab to a particular outlet. For example:
```

```ts
hookTab(
    [
        {
          label: gettext('Import / Export'),
          icon: 'input-output',
          priority: 5,
          path: [
            {
              outlets: {
                'dashboard-details': 'advanced'
              }
            }
          ],
          tabsOutlet: 'dashboardTabs'
        }
      ];
),
hookRoute([
  {
    path: 'advanced',
    loadComponent: () => import(...),
    outlet: 'dashboard-details',
    context: ViewContext.Dashboard
  }
])
```

> It will add additional tab to dashboard settings, and hook route enables it to show view for this tab.

Reviewer: ```suggestion
This will add an additional tab to the dashboard settings, and the hook route allows to display a view for this tab.
```

> This is breaking change, as each component that uses context dashboard need to have rootContext: ViewContext.Dashboard in route definition to make these settings tabs and views visible (even if no new tab was added), e.g.

Reviewer: ```suggestion
This is breaking change, as each component that uses context dashboards must have ´rootContext: ViewContext.Dashboard´ in the route definition to make these settings tabs and views visible (even if no new tab was added). For example:
```

---
date: 2025-04-16

> title: Cross application dashboard copy/paste functionality

Reviewer: ```suggestion
title: Support for dashboard copy/paste functionality across applications
```

> title: Support for dashboard copy/paste functionality across applications

Reviewer 2: ```suggestion
title: Support for cross application dashboard import/export functionality
```

Reviewer 2: Suggesting minor change in the feature title

product_area: Application enablement & solutions
change_type:
  - value: change-inv-3bw8e
    label: Announcement
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-62313, MTM-62316
version: 1022.0.0
---

> In an upcoming version the dashboard settings will get a new tab "Import/export", which allows to export dashboard to json file, import dashboard from previously exported json and edit dashboard object in editor. It as provided as a 'self-optional' plugin to Cockpit, therefore it has to be installed explictly.

Reviewer: ```suggestion
In an upcoming version a new **Import/Export** tab will be added to the dashboard settings, which allows to export dashboards to JSON files, import dashboards from previously exported JSON files and edit the dashboard  in an editor. It is provided as a "self-optional" plugin to the Cockpit application, thus it has to be installed explicitly.
```

> In an upcoming version a new **Import/Export** tab will be added to the dashboard settings, which allows to export dashboards to JSON files, import dashboards from previously exported JSON files and edit the dashboard in an editor. It is provided as a "self-optional" plugin to the Cockpit application, thus it has to be installed explicitly.

Reviewer: ```suggestion
In an upcoming version a new **Import/Export** tab will be added to the dashboard settings, which allows to export dashboards to JSON files, import dashboards from previously exported JSON files and edit the dashboard in an editor. This new functionality allows to copy dashboards across applications. It is provided as a "self-optional" plugin to the Cockpit application, thus it has to be installed explicitly.
```

> In an upcoming version a new **Import/Export** tab will be added to the dashboard settings, which allows to export dashboards to JSON files, import dashboards from previously exported JSON files and edit the dashboard  in an editor. It is provided as a "self-optional" plugin to the Cockpit application, thus it has to be installed explicitly.

Reviewer: ```suggestion
In an upcoming version a new **Import/Export** tab will be added to the dashboard settings, which allows to export dashboards to JSON files, import dashboards from previously exported JSON files and edit the dashboard in an editor. It is provided as a "self-optional" plugin to the Cockpit application, thus it has to be installed explicitly.
```

> Dashboard setting component will be refactored to use secondary router outlet in order to make these kind of views hookable.

Reviewer: ```suggestion
The dashboard setting component will be refactored to use a secondary router outlet in order to make these type of views hookable.
```

This approach allows to hook a new tab to a particular outlet. For example:
```ts
hookTab(
    [
        {
          label: gettext('Import / Export'),
          icon: 'input-output',
          priority: 5,
          path: [
            {
              outlets: {
                'dashboard-details': 'advanced'
              }
            }
          ],
          tabsOutlet: 'dashboardTabs'
        }
      ];
),
hookRoute([
  {
    path: 'advanced',
    loadComponent: () => import(...),
    outlet: 'dashboard-details',
    context: ViewContext.Dashboard
  }
])
```

> This will add an additional tab to the dashboard settings, and the hook route allows to display a view for this tab.

Reviewer: ```suggestion
In this example an additional tab is added to the dashboard settings, and the hook route allows to display a view for this tab.
```
